### PR TITLE
SUBMISSION: My submission to the bash scripting challenge

### PR DESCRIPTION
**Pull request by 5Q1VM5bgef**

### CODE REVIEW

**Comment by 5Q1VM5bgef** [comment]
Hi! Thank you for the challenge!

I've read this book from humble bundle on linux from few months back called "Learning the Bash Shell" by Cameron Newman. Decided to take a script from there as a foundation and enhance it a little to fit for the challenge.

---
Walkthrough:

image

1. Executing script without passing any arguments. Errors and displays usage info.
1. Running and passing giphy.gif, see it converted to giphy.new.jpg. This command uses two flags:
      * -w for changing border width
      * -c for changing border color

File: convert_image.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+cat<<EOF
+Usage: $0 [-S] [-s N] [-w N] [-c S] imagefile...
+
+Options:
+  -S          Change sharpness of a resulting image
+  -s N        Change size of a resulting image
+  -w N        Change width of a border of a resulting image
+  -c S        Change color of a border of a resulting image
+  -h          Show help/usage
+EOF
```
Comment (Sending-Grounds) [inline convert_image.sh]: This is really clean, I like it! The only thing I might suggest is that you add just a one-line or two on what the script does right above the usage.

File: convert_image.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+main() {
+    # Trap signals
+    trap 'handle_signals' SIGINT SIGTERM
+
+    # Set up default width and color
+    width=1
+    colour='-color grey'
+
+    # Initialise the pipeline components
+    standardise=' | pnmtojpeg -quiet'
```
Comment (Sending-Grounds) [inline convert_image.sh:74]: One thing you might consider is having some form of a `setup()` func where you can do some of the initial setup, or in this case you're using `pnmnlfilt` and associated utilities but we don't have a check to see if they are installed or in our PATH. Bit of a nicety, but it makes it a better user experience.

Basically just to tackle the below for them/yourself.

```bash
[Sending_Grounds@HomeBase ~]$ type pnmnlfilt
-bash: type: pnmnlfilt: not found
[Sending_Grounds@HomeBase ~]$ echo $?
1
[Sending_Grounds@HomeBase ~]$ sudo yum whatprovides pnmnlfilt
...<TRUNCATED>...
netpbm-progs-10.79.00-7.amzn2.x86_64 : Tools for manipulating graphics files in netpbm supported formats
Repo        : amzn2-core
Matched from:
Filename    : /usr/bin/pnmnlfilt

[Sending_Grounds@HomeBase ~]$ sudo yum install netpbm-progs-10.79.00-7.amzn2.x86_64 -y
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
Resolving Dependencies
--> Running transaction check
...<TRUNCATED>...
Complete!
[Sending_Grounds@HomeBase ~]$ type pnmnlfilt
pnmnlfilt is /usr/bin/pnmnlfilt
[Sending_Grounds@HomeBase ~]$ echo $?
0
```

File: convert_image.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+            *.jpg ) convert='jpegtopnm -quiet' ;;
+
+                * ) echo "$0: Unknown filetype '${filename##*.}'"
+                    exit 1;;
+        esac
+
+        outfile=${filename%.*}.new.jpg
+
+        eval $convert $filename $scale $border $sharpness $standardise > $outfile
```
Comment (Sending-Grounds) [inline convert_image.sh]: The only call out would be to try and stick with `${variable}` structure that will help prevent whitespace and variable expansion issues in the future, especially if your var is user input and includes something like unexpected whitespace, it'll cause your script to break in unpredictable ways

**Review by Sending-Grounds — APPROVED** [review]
Awesome script! Love the idea of a CLI tool to do some image manipulation. It's outside of my wheelhouse so I'll have to dig in more. Also really appreciated the submission!

**Review by Sending-Grounds — APPROVED** [review]
Ship Ittttttt

**Pull request merged**